bindings/c/src/types.rs

```diff
@@ -97,6 +97,57 @@ impl From<&mut libsql_connection> for libsql_connection_t {
     }
 }
 
+pub struct stmt {
+    pub sql: String,
+    pub params: Vec<libsql::Value>,
+}
+
+pub struct libsql_stmt {
+    pub stmt: stmt,
+}
+
+#[derive(Clone, Debug)]
+#[repr(transparent)]
+pub struct libsql_stmt_t {
+    ptr: *const libsql_stmt,
+}
+
+impl libsql_stmt_t {
+    pub fn null() -> libsql_stmt_t {
+        libsql_stmt_t {
+            ptr: std::ptr::null(),
+        }
+    }
+
+    pub fn is_null(&self) -> bool {
+        self.ptr.is_null()
+    }
+
+    pub fn get_ref(&self) -> &stmt {
+        &unsafe { &*self.ptr }.stmt
+    }
+
+    #[allow(clippy::mut_from_ref)]
+    pub fn get_ref_mut(&self) -> &mut stmt {
```

Comment:
I think this needs to be marked unsafe, converting a &_ to a &mut _ is a UB and violates LLVM aliasing. So this likely should return the raw *mut pointer as well and not convert it back into a safe rust &mut

Reply:
Does that mean all the other get_ref_mut methods in this file are wrong too?